python/cudf/cudf/core/frame.py

```diff
@@ -139,6 +153,29 @@ def _from_data(
         Frame.__init__(obj, data, index)
         return obj
 
+    @classmethod
+    def _from_maybe_indexed_columns(
+        cls,
+        columns: List[ColumnBase],
+        column_names: List[str],
+        index_names: Optional[List[str]] = None,
+    ):
+        index = None
+        if index_names is not None:
+            index_ids = list(range(len(index_names)))
+            # First construct the index, if any
+            index = cudf.core.index._index_from_data(
+                dict(zip(index_ids, columns))
+            )
```

Comment:
Are the `index_names` supposed to be used here? It looks like only the length of `index_names` is currently being used in this method.

Reply:
Oh yeah, good catch!

Reply:
So I figured the `index_name` argument for `_index_from_data` only accepts a single level index name, which has some conflicts when the `index_names` has multiple levels. I plan to add this fix in a future PR. For now, I would implement the below logic in `_from_maybe_indexed_columns`.